### PR TITLE
expose MyAvatar.xxxHandPose to JS

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 71dfff7c3579c9395c03c1d356cd6535c011660b available at: https://gist.github.com/7712c9eb274c9add98fb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4a58eeb810bdee721e44842477b15c5eb832e30c available at: https://gist.github.com/7712c9eb274c9add98fb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.dmg

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -596,6 +596,34 @@ glm::vec3 MyAvatar::getRightHandTipPosition() const {
     return palmData ? palmData->getTipPosition() : glm::vec3(0.0f);
 }
 
+controller::Pose MyAvatar::getLeftHandPose() const {
+    const int LEFT_HAND = 0;
+    auto palmData = getActivePalm(LEFT_HAND);
+    return palmData ? controller::Pose(palmData->getPosition(), palmData->getRotation(),
+        palmData->getVelocity(), palmData->getRawDeltaRotation()) : controller::Pose();
```
Comment (jherico) [inline interface/src/avatar/MyAvatar.cpp]: Delta rotation isn't being divided by the delta time in the `MyAvatar`, so it's not an angular velocity (radians / second), just a straight rotation.  I don't know that you can reasonably divide a quaternion by a scalar, though.  I believe what Sam does in the hydra code is divide the Euler form by the time and then reconstructs a quaternion from the Euler angles.  You could do the same here by fetching `getRawAngularVelocity` and turning it into a quaternion.

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -596,6 +596,34 @@ glm::vec3 MyAvatar::getRightHandTipPosition() const {
     return palmData ? palmData->getTipPosition() : glm::vec3(0.0f);
 }
 
+controller::Pose MyAvatar::getLeftHandPose() const {
+    const int LEFT_HAND = 0;
+    auto palmData = getActivePalm(LEFT_HAND);
+    return palmData ? controller::Pose(palmData->getPosition(), palmData->getRotation(),
```
Comment (jherico) [inline interface/src/avatar/MyAvatar.cpp]: A palmDataToPose function here would reduce duplication.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f90844449d33b0c86bb355b2dbd8dffc22290665 available at: https://gist.github.com/7712c9eb274c9add98fb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6164.dmg

**Pull request merged**